src/schema/cell.rs

```diff
@@ -1126,6 +1143,20 @@ where
                         None => Ok(None),
                     }
                 }
+                DataType::Date32 => match self.get_primitive_value::<Date32Array>(index)? {
```

Comment:
Could you make this follow the same pattern as `self.get_timestamp_value` above?